contextproject-ms3/src/main/java/contextproject/models/Key.java

```diff
@@ -1,166 +1,181 @@
+        normalizedKeyFlag = "A";
         break;
 
       case "Bbm" :
-        keyNumberPart = 3;
-        keyCharPart = "A";
+        normalizedKeyNumber = 3;
+        normalizedKeyFlag = "A";
         break;
 
       case "Bm" :
-        keyNumberPart = 10;
-        keyCharPart = "A";
+        normalizedKeyNumber = 10;
+        normalizedKeyFlag = "A";
         break;
 
       case "Cm" :
-        keyNumberPart = 5;
-        keyCharPart = "A";
+        normalizedKeyNumber = 5;
+        normalizedKeyFlag = "A";
         break;
 
       case "Dbm" :
-        keyNumberPart = 12;
-        keyCharPart = "A";
+        normalizedKeyNumber = 12;
+        normalizedKeyFlag = "A";
         break;
 
       case "Dm" :
-        keyNumberPart = 7;
-        keyCharPart = "A";
+        normalizedKeyNumber = 7;
+        normalizedKeyFlag = "A";
         break;
 
       case "Ebm" :
-        keyNumberPart = 2;
-        keyCharPart = "A";
+        normalizedKeyNumber = 2;
+        normalizedKeyFlag = "A";
         break;
 
       case "Em" :
-        keyNumberPart = 9;
-        keyCharPart = "A";
+        normalizedKeyNumber = 9;
+        normalizedKeyFlag = "A";
         break;
 
       case "Fm" :
-        keyNumberPart = 4;
-        keyCharPart = "A";
+        normalizedKeyNumber = 4;
+        normalizedKeyFlag = "A";
         break;
 
       case "Gbm" :
-        keyNumberPart = 11;
-        keyCharPart = "A";
+        normalizedKeyNumber = 11;
+        normalizedKeyFlag = "A";
         break;
 
       case "Gm" :
-        keyNumberPart = 6;
-        keyCharPart = "A";
+        normalizedKeyNumber = 6;
+        normalizedKeyFlag = "A";
         break;
 
       case "Abm" :
-        keyNumberPart = 1;
-        keyCharPart = "A";
+        normalizedKeyNumber = 1;
+        normalizedKeyFlag = "A";
         break;
 
       default :
-        keyNumberPart = 0;
-        keyCharPart = null;
-        log.warn("Song has no key in ID3 information.");
-        break;
+        boolean foundNormalizedKey = parseNormalizedString(keyString);
+        if (!foundNormalizedKey) {
+          throw new IllegalArgumentException("Key cannot be detected");
+        }
+    }
+  }
+  
+  /**
+   * Try to parse the key string if it is normalized already.
+   * @param keyString The given key as text
+   * @return Boolean whether the parsing succeeded or not
+   * @throws NumberFormatException When matching fails
+   */
+  private boolean parseNormalizedString(String keyString) throws NumberFormatException {
+    Pattern pattern = Pattern.compile("([0-9]{1,2})([AB])");
+    Matcher matcher = pattern.matcher(keyString);
+    if (matcher.matches()) {
+      int number = Integer.parseInt(matcher.group(1));
+      String flag = matcher.group(2);
+      if (number > 0 && number < 13 && (flag.equals("A") || flag.equals("B"))) {
+        this.normalizedKeyNumber = number;
+        this.normalizedKeyFlag = flag;
+        return true;
+      }
     }
+    return false;
   }
```

Comment:
If this method fails, line 154-155 get executed, which in turn throws an exception that gets logged.